aioxmpp/avatar/service.py

```diff
@@ -509,19 +566,20 @@ def publish_avatar_set(self, avatar_set):
 
         id_ = avatar_set.png_id
 
-        yield from self._pubsub.publish(
-            None,
-            namespaces.xep0084_data,
-            avatar_xso.Data(avatar_set.image_bytes),
-            id_=id_
-        )
+        with (yield from self._lock):
```

Comment:
This still lacks the race condition fix if we race with other clients, right? Do you plan to do this in this PR?